src/Semantic/Git.hs

```diff
@@ -15,14 +17,21 @@ module Semantic.Git
   , parseEntry
   ) where
 
-import Control.Monad.IO.Class
-import Data.Attoparsec.Text   (Parser)
-import Data.Attoparsec.Text   as AP
-import Data.Char
-import Data.Either            (fromRight)
-import Data.Text              as Text
-import Shelly                 hiding (FilePath)
-import System.IO              (hSetBinaryMode)
+import Prologue
+
+import           Control.Monad.IO.Class
+import           Data.Attoparsec.ByteString.Char8 (Parser)
+import           Data.Attoparsec.ByteString.Char8 as AP
+import qualified Data.Attoparsec.ByteString.Streaming as AP.Stream
+import qualified Data.ByteString.Char8 as BC
+import qualified Data.ByteString.Streaming.Char8 as ByteStream
```

Comment:
The `.Char8` modules are kinda dangerous if there’s any chance you could be doing something in utf8. Is conversion from e.g. `Text` prohibitively expensive?

Reply:
I can change this to use the Word8 equivalents.

Reply:
I feel like converting to Text for parsing purposes, only to stick it back into a ByteString, would kill a lot of the speed that this patch gives. Maybe it’s a sign we shouldn’t do it? I do not know.

Reply:
> I don’t think we would be getting any non-UTF8 pathnames, since the system encoding is presumably set to UTF-8, no? Is this naïve? We could, and perhaps should, rejigger this so that it operates on `Word8` values—I think this is possible, since the only queries we make of the input are testing it to see if a value is alphanumeric/a space/a tab/NUL.

Are we 100% sure that none of those bytes can occur as part of multi-byte sequences or something?

Reply:
Adding a Unicode-path-name test to this broke not just this branch, but master 😱 Looks like we’ve never been handling UTF-8 stuff correctly, or if we have, it’s been an artifact of luck.